### PR TITLE
[aaf_logging][aaf_monitor_logs] Changed to run logging all the time, added log monitoring

### LINKED ISSUE #286: logging
check the logging is working as it should and is started in the main startup script. monitor daily!

### PR DESCRIPTION
**Pull request by nilsbore**
This adds the bash script to log all the topics to its own tmux pane and the idea is that it will be running logging of these topics all the time. There is also the new package `aaf_monitor_logs`, which publishes on the topics `/monitored_logs` every 30 s. The message is a json dictionary with the topics and how many messages were logged in the last 30 s together with the total number of messages in the data base.

The reason for not having logging start and stop at the beginning and end of each day is two-fold. First, we cannot start one node logging all the topics in a one launch file, detailed in https://github.com/strands-project/mongodb_store/issues/116 . The other possibility is starting one node separately for each topic, within a large launch file. That was found to be too taxing on memory consumption, but one example of such a launch file is included in this pull request, see `topics_launch.launch`. In general, we need to start things in launch files for it to work with the `axlaunch_server`.

This should fix https://github.com/strands-project/aaf_deployment/issues/286 .

### CODE REVIEW

File: aaf_monitor_logs/package.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0"?>
+<package>
+  <name>aaf_monitor_logs</name>
+  <version>0.1.20</version>
+  <description>A package to support MongoDB-based storage and analysis for data from a ROS system, eg. saved messages, configurations etc</description>
```
Comment (marc-hanheide) [inline aaf_monitor_logs/package.xml]: @nilsbore is this the right description or copy and paste?

File: aaf_monitor_logs/package.xml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  <maintainer email="cburbridge@gmail.com">Chris Burbridge</maintainer>
```
Comment (marc-hanheide) [inline aaf_monitor_logs/package.xml]: Also all of this looks like wrong copy/paste. Can @nilsbore please fix this first?

File: aaf_monitor_logs/scripts/monitor_logs.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+        self.topics_map = {}
+
+        self.pub = rospy.Publisher('monitored_logs', String, queue_size=10)
+        self.timer = rospy.Timer(rospy.Duration(30), self.timer_callback)
+
+    def timer_callback(self, time):
+        for topic in self.topic_list:
+            #print "Getting messages for " + topic
+            nbr_messages = bridge.find_nbr_entries("roslog", dc_util.topic_name_to_collection_name(topic))
```
Comment (marc-hanheide) [inline aaf_monitor_logs/scripts/monitor_logs.py]: should this not be `self` rather than the global `bridge`?

Comment (marc-hanheide) [inline aaf_monitor_logs/scripts/monitor_logs.py]: Also, is the collection name always "roslog"?

File: aaf_monitor_logs/scripts/monitor_logs.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import rospy
+import mongodb_store.util as dc_util
```
Comment (marc-hanheide) [inline aaf_monitor_logs/scripts/monitor_logs.py:5]: I believe `mongodb_store` is not listed in `package.xml` as a dependency

File: aaf_monitor_logs/package.xml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  <author>Nick Hawes</author>
+
+  <url type="website">http://www.ros.org/wiki/mongodb_store</url>
+
+  <license>MIT</license>
+
+  <buildtool_depend>catkin</buildtool_depend>
+
+  <build_depend>rospy</build_depend>
+  <build_depend>roscpp</build_depend>
```
Comment (marc-hanheide) [inline aaf_monitor_logs/package.xml]: I don't think this needs roscpp, also check the other deps, please

File: aaf_monitor_logs/cmake/FindMongoClient.cmake [hunk 6]
```diff
@@ -0,0 +1,78 @@
+# - Find MongoClient; NOTE: this is specific to warehouse_ros!
```
Comment (marc-hanheide) [inline aaf_monitor_logs/cmake/FindMongoClient.cmake]: again, this file is not needed, is it? There is no CPP code in here... Also `CMakeLists.txt` should be fixed to only have the right deps in it.

File: aaf_monitor_logs/scripts/monitor_logs.py (hunk 3 above)
Comment (nilsbore) [inline aaf_monitor_logs/scripts/monitor_logs.py]: It should indeed be `self`. Not sure why this works but it does. I'll fix it. As far as roslog goes, the logged topic and monitored topics are all in roslog.

**Comment by nilsbore** [comment]
@marc-hanheide I addressed your comments and checked that it still works.

**Comment by marc-hanheide** [comment]
great, thanks!

**Pull request merged**